gulpfile.js

```diff
@@ -72,21 +72,26 @@ function test() {
 gulp.task('build-test', ['build'], test);
 gulp.task('test', test);
 
-gulp.task('checkImports', function (cb) {
-    var checkProcess = child_process.fork(path.join(__dirname, "tools", "checkCasing.js"),
-        {
-            cwd: path.resolve(__dirname, "src"),
-            stdio: "inherit"
+function defineCustomVerification(name, pathInTools, errorMessage) {
+    gulp.task(name, function (cb) {
+        var checkProcess = child_process.fork(path.join(__dirname, "tools", pathInTools),
+            {
+                cwd: path.resolve(__dirname, "src"),
+                stdio: "inherit"
+            });
+        checkProcess.on("error", cb);
+        checkProcess.on("exit", function (code, signal) {
+            if (code || signal) {
+                cb(new Error(errorMessage));
+            } else {
+                cb();
+            }
         });
-    checkProcess.on("error", cb);
-    checkProcess.on("exit", function (code, signal) {
-        if (code || signal) {
-            cb(new Error("Mismatches found in import casing"));
-        } else {
-            cb();
-        }
     });
-});
+}
+
+defineCustomVerification('checkImports', "checkCasing.js", "Mismatches found in import casing");
```

Comment:
It's nice that we are reusing code here but we should have it the other way around in order not to confuse users. I would call the common function from two declared gulp tasks. It's easier to understand and follow if we define tasks the same way.